agent/configuration-build-derive/src/derive.rs

```diff
@@ -0,0 +1,431 @@
+use proc_macro2::{Ident, Span, TokenStream};
```

Comment:
A docstring at the top of this module or in `lib.rs` explaining what this is and how to use it would be very helpful.